src/Shabalov/Task-5/main.cpp

```diff
@@ -0,0 +1,154 @@
+#include <QImage>
+#include <QColor>
+#include <QOpenGLFunctions>
+#include <QVector3D>
+#include <cmath>
+#include "Ray.h"
+#include "Sphere.h"
+#include "Light.h"
+#include "Plane.h"
+#include <memory>
+
+#ifndef M_PI
+    #define M_PI 3.14159265358979323846
+#endif
+
+constexpr auto WIDTH = 1280;
+constexpr auto HEIGHT = 960;
+constexpr auto FOV = M_PI/3.;
+constexpr auto MAX_REFLECTIONS = 4;
+constexpr auto BACKGROUND_COLOR = QVector3D(0.2f, 0.7f, 0.8f);
+
+
+QVector3D mix(const QVector3D &a, const QVector3D &b, const float k){
```

Comment:
```suggestion
[[nodisacrd]] auto mix(const QVector3D &a, const QVector3D &b, const float k){
```
и для остальных функций